src/song_utils.py

```diff
@@ -8,11 +9,55 @@
 from mutagen.id3 import ID3FileType, PictureType
 from mutagen.ogg import OggFileType
 from mutagen.wave import WAVE
-from nextcord import File
+from nextcord import File, Embed, Message, Attachment, User, Member
+from nextcord.utils import escape_markdown
 from PIL import Image, UnidentifiedImageError
 
 import config
 
+def embed_song(submit_message: Message | str, attachment_filepath: str, attachment: Attachment, user: User | Member, random_emoji=None) -> Embed:
+    
+    if isinstance(submit_message, Message):
+        message: str = submit_message.content
+    else:
+        message: str = submit_message
+        
+    emoji = random_emoji if random_emoji != None else random.choice(config.emoji_list)
+    
+    # Build and send "Now Playing" embed
+    if isinstance(submit_message, Message) == False:
```

Comment:
This function duplicates a lot of code, it's almost two copies of the same piece of code with a few if statements. You should try to find some better ways to combine similar lines.

I also think you should not change things in the format like Playing -> Previewing and removing the jump. I realize that a jump link doesn't make sense, but I think it should still be there and point to some sensible default URL. Things should be as visually identical as possible